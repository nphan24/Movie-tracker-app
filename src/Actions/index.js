export const postMovies = movies => ({
  type: 'POST_MOVIES',
  movies
});

export const addUser = user => ({
  type: 'ADD_USER',
  user
});

export const signInAction = user => ({
  type: 'SIGN_IN',
  user
});

export const validSignIn = error => ({
  type: 'VALID_SIGNIN',
  error
});

export const invalidSignIn = error => ({
  type: 'INVALID_SIGNIN',
  error
});

export const logoutUser = () => ({
  type: 'LOGOUT_USER'
});
