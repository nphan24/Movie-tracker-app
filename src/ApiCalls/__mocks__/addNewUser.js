export const addNewUser = jest.fn();
