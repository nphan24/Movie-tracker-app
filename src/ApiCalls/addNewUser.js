export const addNewUser = async ({ name, email, password }) => {
  try {
    const response = await fetch("/api/users/new", {
      method: "POST",
      body: JSON.stringify({
        name: name,
        email: email,
        password: password
      }),
      headers: {
        "Content-Type": "application/json"
      }
    });
    const data = await response.json();
    if (data.error) {
      throw new Error(data.error);
    } else {
      return data.id;
    }
  } catch (error) {
    throw new Error(`${email} already exists. Please enter a new email`);
  }
};
