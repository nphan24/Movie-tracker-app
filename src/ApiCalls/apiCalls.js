import apiKey from './apiKey';

export const fetchData = async () => {
  try {
  const response = await fetch(
    `https://api.themoviedb.org/3/movie/upcoming?api_key=${apiKey}`
  );
  const data = await response.json();
  return data.results
  } catch (error) {
    throw 'error'
  }
} 

export const getMovies = async () => {
  const movieData = await fetchData();
  console.log(movieData);
  // const cleanedMovies = cleanMovies(movieData);
  // return cleanedMovies
}

export const cleanMovies = (movieData) => {
  const cleanedMovies = movieData.map(movie => {

  })
}
 
