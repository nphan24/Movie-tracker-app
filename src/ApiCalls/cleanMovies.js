export const cleanMovies = movieData => {
  const cleanedMovies = movieData.map(movie => {
    const {
      title,
      overview,
      release_date,
      vote_average,
      poster_path,
      popularity
    } = movie;
    return {
      title,
      movie_id: Math.floor(Math.random() * 100000),
      overview: `Summary: ${overview}`,
      release_date: `Release Date: ${release_date}`,
      vote_average: `Rating: ${vote_average}`,
      poster_path: poster_path
    };
  });
  return cleanedMovies;
};
