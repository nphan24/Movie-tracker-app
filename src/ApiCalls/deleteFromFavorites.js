export const deleteFromFavorites = async (movieid, userid) => {
  try {
    await fetch(`api/users/${userid}/favorites/${movieid}`, {
      method: 'DELETE'
    });
  } catch (error) {
    throw new Error(error.message);
  }
};
