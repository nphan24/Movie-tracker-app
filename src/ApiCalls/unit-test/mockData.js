export const mockMovieData = {
  results: [
    {
      original_language: 'en',
      original_title: 'Jumanji: Welcome to the Jungle',
      overview:
        "The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
      popularity: 101.350698,
      poster_path: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
      release_date: '2017-12-09',
      title: 'Jumanji: Welcome to the Jungle',
      video: false,
      vote_average: 6.5,
      vote_count: 3115
    }
  ]
};

export const cleanedMovie = [
  {
    date: 'Release Date: 2017-12-09',
    id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];

export const movieInfo = {
  movie_id: '1',
  user_id: '24',
  title: 'hello',
  poster_path: '/hello',
  release_date: '04-01-18',
  vote_average: '-5',
  overview: 'Its not real'
};

export const mockUser = {
  id: '5',
  name: 'r',
  email: 'r@r.r',
  password: 'r'
};

export const mockFavorite = {
  date: 'Release Date: 2017-12-09',
  movie_id: 101.350698,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview:
    "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
  title: 'Jumanji: Welcome to the Jungle',
  vote: 'Rating: 6.5'
};

export const mockRemainingFavorite = {
  date: 'Release Date: 2018-10-5',
  movie_id: 10,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview: 'Summary: The tables are turned as four teenagers.',
  title: 'Fantastic Fours',
  vote: 'Rating: 2.5'
};

export const mockFavoritesArray = [
  {
    date: 'Release Date: 2018-10-5',
    movie_id: 10,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview: 'Summary: The tables are turned as four teenagers.',
    title: 'Fantastic Fours',
    vote: 'Rating: 2.5'
  },
  {
    date: 'Release Date: 2017-12-09',
    movie_id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];
