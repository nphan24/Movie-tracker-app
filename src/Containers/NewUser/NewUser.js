import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addNewUser } from '../../ApiCalls/addNewUser';
import * as Actions from '../../Actions';
import { NavLink, withRouter } from 'react-router-dom';

export class NewUser extends Component {
  constructor() {
    super();
    this.state = {
      name: '',
      email: '',
      password: ''
    };
  }

  handleInput = event => {
    const { name, value } = event.target;

    this.setState({
      [name]: value
    });
  };

  handleSubmit = async (event) => {
    const { name, email } = this.state
    const id = await addNewUser({ ...this.state });
    this.props.postNewUser({name, email, id});
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <NavLink to="/signIn">Login</NavLink>
        <label>
          <input
            type="text"
            placeholder="Name"
            value={this.state.name}
            name="name"
            onChange={this.handleInput}
          />
        </label>
        <label>
          <input
            type="email"
            placeholder="Email"
            value={this.state.email}
            name="email"
            onChange={this.handleInput}
          />
        </label>
        <label>
          <input
            type="password"
            placeholder="Password"
            value={this.state.password}
            name="password"
            onChange={this.handleInput}
          />
        </label>
        <NavLink to='/' onClick={this.handleSubmit}>
          <button>Submit</button>
        </NavLink>
      </form>
    );
  }
}

const mapDispatchToProps = (dispatch) => ({
  postNewUser: (user) => dispatch(Actions.addUser(user))
})

export default withRouter(connect(null, mapDispatchToProps)(NewUser));
