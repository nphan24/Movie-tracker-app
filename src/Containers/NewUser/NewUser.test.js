import React from 'react';
import { NewUser, mapDispatchToProps } from "./NewUser";
import { shallow } from 'enzyme';
import { addNewUser } from "../../ApiCalls/addNewUser";
import * as Actions from "../../Actions";
import * as mock from "../../ApiCalls/unit-test/mockData";
jest.mock('../../ApiCalls/addNewUser.js');

describe('Login', () => {
  let wrapper;
  let mockpostNewUser;
  let mockvalidSignIn;
  let mockinvalidSignIn;
  beforeEach(() => {
    mockpostNewUser = jest.fn();
    mockvalidSignIn = jest.fn();
    mockinvalidSignIn = jest.fn();
    wrapper = shallow(
      <NewUser 
        postNewUser={mockpostNewUser}
        validSignIn={mockvalidSignIn}
        invalidSignIn={mockinvalidSignIn}
      />);
  });
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state on input change', () => {
    const mockEvent = { target: { value: 'ahhhhh', name: 'name' } };
    wrapper.instance().handleInput(mockEvent);
    expect(wrapper.state('name')).toEqual('ahhhhh');
  });

  it('should add a new user', async () => {
    await wrapper.instance().handleSubmit();
    expect(addNewUser).toHaveBeenCalled();
  });

  it('should postNewUser', async () => {
    await wrapper.instance().handleSubmit();
    expect(mockpostNewUser).toHaveBeenCalled();
  });

  it("should postNewUser", async () => {
    await wrapper.instance().handleSubmit();
    expect(mockvalidSignIn).toHaveBeenCalled();
  });

  it("should run invalidSignIn on error", async () => {
    mockpostNewUser = jest.fn().mockImplementation(() => {
      throw new Error("Error");
    });
    wrapper = shallow(
      <NewUser
        postNewUser={mockpostNewUser}
        invalidSignIn={mockinvalidSignIn} 
      />);
    await wrapper.instance().handleSubmit();
    expect(mockinvalidSignIn).toHaveBeenCalled();
  });
});

describe("mapDispatchToProps", () => {
  it("should call dispatch with correct params on postNewUser", () => {
    const mockDispatch = jest.fn();
    const user = mock.mockUser;
    const expected = Actions.addUser(user);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.postNewUser(user);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it("should call dispatch with correct params on invalidSignIn", () => {
    const mockDispatch = jest.fn();
    const error = "error";
    const expected = Actions.invalidSignIn(error);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.invalidSignIn(error);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });

  it("should call dispatch with correct params on validSignIn", () => {
    const mockDispatch = jest.fn();
    const valid = "true";
    const expected = Actions.validSignIn(valid);
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.validSignIn(valid);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});


