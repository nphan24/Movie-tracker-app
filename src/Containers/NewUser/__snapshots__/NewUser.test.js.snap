// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewUser />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/signIn"
>
          Login
</NavLink>,
        <label>
          <input
                    name="name"
                    onChange={[Function]}
                    placeholder="Name"
                    type="text"
                    value=""
          />
</label>,
        <label>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="email"
                    value=""
          />
</label>,
        <label>
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
</label>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          onClick={[Function]}
          to="/"
>
          <button>
                    Submit
          </button>
</NavLink>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Login",
          "to": "/signIn",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="email"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email",
            "type": "email",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "password",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            Submit
</button>,
          "onClick": [Function],
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/signIn"
>
            Login
</NavLink>,
          <label>
            <input
                        name="name"
                        onChange={[Function]}
                        placeholder="Name"
                        type="text"
                        value=""
            />
</label>,
          <label>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="email"
                        value=""
            />
</label>,
          <label>
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
            />
</label>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            onClick={[Function]}
            to="/"
>
            <button>
                        Submit
            </button>
</NavLink>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Login",
            "to": "/signIn",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="name"
              onChange={[Function]}
              placeholder="Name"
              type="text"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "placeholder": "Name",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="email"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              Submit
</button>,
            "onClick": [Function],
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
